File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/state/orderstate/OrderDeliverState.java
// OrderDeliverState.java

package com.bigboxcode.designpattern.state.orderstate;

public class OrderDeliverState extends OrderState {
    public OrderDeliverState(OrderContext context) {
        super(context);
    }

    @Override
    public void process() {
        // Write code to process the order
        System.out.println("Delivering the order");

        context.setState(context.getOrderReceiveState());
    }
}

File: design-patterns/php/state/order/OrderDeliverState.php
<?php
// OrderDeliverState.php

namespace BigBoxCode\DesignPattern\State\Order;


class OrderDeliverState extends OrderState {
    public function __construct(OrderContext $context) {
        parent::__construct($context);
    }

    public function process(): void {
        // Write code to process the order
        echo "Delivering the order\n";

        $this->context->setState($this->context->getOrderReceiveState());
    }
}